### ISSUE TITLE
Feature request: Multiple one-way categorical frequencies/proportions in wide format

### ISSUE THREAD

**Issue by sda030** [ddsjoberg/gtsummary#1290]
*TL;DR: Jump to table at the bottom.*

**Problem:**
Creating a single table (tibble or HTML-output) with multiple categorical univariates, like that often occuring when dealing with survey data, is tedious in R compared to commercial software like SPSS and Stata. (In fact, it is embarrassing when I try to convince colleagues to shift to R that there are no straightforward base or tidyverse functions). 

***Examples of unsatisfactory solutions:***
Sure, there are plenty of ways of creating crosstabs with one or two categorical variables:
```library(dplyr)
mtcars %>%
  dplyr::count(cyl, name = "p") %>%
  dplyr::mutate(p = p/sum(p, na.rm=TRUE))
```

The srvyr-package is fine, and a lot better than alternatives if one has survey design features. But drawbacks regarding the unintuitive way of dealing with a list object rather than a data frame, and the fact that it serves as a translator between tidyverse and `survey` (but that is another story). Still not multiple univariates though.
```mtcars %>% 
  srvyr::as_survey() %>%
  dplyr::group_by(cyl) %>%
  dplyr::summarize(pct = srvyr::survey_prop())
```

The quite popular janitor package has the tabyl()-functions for making tabulations of categorical variables, and provides one-way, two-way and three-way crosstabs easily. Though I think these functions are superseeded by gt and gtsummary, and still no multiple univariates.
```mtcars %>%
  janitor::tabyl(cyl)
```

gtsummary is a lot prettier, more flexible and more intuitive for one-way and two-way frequency/proportion tables. But still no multiple univariates. In fact, the complex object returned from the function makes it even more difficult to process the output with tidyr::pivot_wider() further down the pipeline. 
```mtcars %>%  
  gtsummary::tbl_summary(include = cyl)
```

***Solution:***
However, none of the above-mentioned packages provide intuitive functions for dealing with data from questionnaires where there are multiple questions in a single instrument with the same response options (typically Likert). Below is what my colleagues want. A one-liner that for a set of similar categorical variables outputs percentages/counts/proportions in wide format.
```library(dplyr)  
ex_survey1 %>%  
  select(matches("b_")) %>%  # All three columns in this dataset happens to belong to the same instrument.
  tbl_summary_multi_uni()  # Can have all the typical arguments on missingness, labels, etc.
#>  var                            variable_label    Not at all A bit A lot  
#>  b_1  How much do you like living in - Bejing         55    37     8  
#>  b_2  How much do you like living in - Brussels       40    48    12  
#>  b_3  How much do you like living in - Budapest       45    42    13  
```
Notice that in my own function, the output tibble includes the variable label, if it exists. This tibble is now ready for prettification with gt, ggplot2, or similar.

I also have started devleoping another convenience function for when one has multiple of these instruments (i.e. sets of categorical variables with the same response options) in a data set. Under certain conditions (which one specifies), one can "split" the dataset colwise accordingly, run the  tbl_summary_multi_uni() function above, and then output either a list of summary tables, or output like gtsummary does with include=c(b_1, b_2, b_3), i.e. listed downwards.

I want these functions in a existing, relevant and the most promising package so I have scoured CRAN for a suitable home. Let me know if you want these orphans. They will not introduce any new dependencies/imports. However, given the complexity of

**Comment by ddsjoberg** [comment]
@sda030 thank you for the detailed post.

We have a draft of a function like this available in the bstfun package. The bstfun pkg often serves as a sandbox/dev area for functions that are eventually migrated to gtsummary. I think this function will make your life easier.
https://www.danieldsjoberg.com/bstfun/reference/tbl_likert.html

**Comment by sda030** [comment]
Amazing! You should have told me before I wasted hours developing something inferior to this. ;) 
When will it be migrated? In August I aim to convince 12 non-R users to convert to the Right way of life and want to have all the example code ready for them. ;)

P.S. The only limitation I can see compared to my function is the inability to combine variable name and variable label (goes for all the tbl_summary functions), which could be fixed with a similar glue::glue() solution you use elsehwhere. But it is such a minor issue that I do not need to wait for that.

**Comment by ddsjoberg** [comment]
I don't have a timeline for its migration at the moment

**Comment by sda030** [comment]
Hi dear @ddsjoberg, would you mind me trying a PR for this (from your bstfun) with some extensive unit tests, within the next month?

**Comment by larmarange** [comment]
Just to be noted: for visualisation, you canalso have a look at gglikert() : https://larmarange.github.io/ggstats/articles/gglikert.html

Regarding `tbl_likert()` it is not yet compatible with survey object but works well with unweighted data frames.

**Comment by larmarange** [comment]
`tbl_likert()` was built around `tbl_summary()` and its ok like that.

It would be easy at a later stage to develop `tbl_svylikert()` using `tbl_svysummary()` instead of `tbl_summary()`
